### PR TITLE
Some Ling Changes

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Regenerative Stasis takes 40s down from 80s
New power for lings: Chameleon Skin. Costs 2 evolution points, uses 25 chems on toggling, and basically gives you the chameleon skin genetics power that lets you fade in gradually
Fleshmend chem cost changed from 25 to 20, restores only a bit of blood (use regen stasis for that) and repeated use of it quickly results in less effective healing
Armblade opening doors takes 10s now down from 15s just like the Jaws of Life. Still plenty of time to get caught with an armblade out.
Anatomic Panacea also gives you Antihol and Mannitol now, curing drunkenness and brain damage
When lings die, they will still generate chems up to half the maximum (so 37.5). I noticed that when lings got into fights, they'd use up all their chemicals fast, and when they finally died they couldn't go into Regen Stasis even if they had 14 chems. So now that they generate when dead you can go into Regen Stasis when you're dead.
I made it impossible to accidentally lose the ability to Revive if you Readapt while in stasis.
As with the Biodegrade PR this is from /tg/

:cl:
add: Chameleon Skin power for changelings. Costs 2 evolution points and lets you toggle the Chameleon gene on and off
balance: Changelings will now regenerate chemicals when dead, up to half the maximum (37.5)
balance: Fleshmend costs 20 chems down from 25. However it no longer restores all blood only a bit of blood volume, and using it multiple times quickly results in less effective healing
balance: Regenerative Stasis takes half the time, 40s instead of 80s
fix: Changelings cannot accidentally lose the ability to revive by readapting in stasis
tweak: Armblade opening doors takes 10s instead of 15s
tweak: Anatomic Panacea cures drunkenness and brain damage too
/:cl:

### CODE REVIEW

**Comment by Rumiluntti** [comment]
Clownlings with tentacles and chameleon skin are now haunting my nightmares.
Thanks.

**Comment by SamHPurp** [comment]
This is an enormous buff to changelings. Gigantic.

**Comment by uraniummeltdown** [comment]
It's not enormous, the main healing power is nerfed and doesn't restore blood anymore. Changelings should not be permadead because of not having enough chemicals to regen. You use up all your chems in combat, and their whole thing is coming back from the dead unless they're completely destroyed. Plus when you revive, you have genetic damage (genomes reassembling cooldown thing) and only 37.5 chemicals to use.

**Comment by granodd** [comment]
That's a lot of buffs. Did Changelings really need all that?

**Comment by Varghh** [comment]
Too many buffs for an antag who is already hella strong

**Comment by FalseIncarnate** [comment]
Chameleon skin is an interesting addition, making it work like the genetic power rather than just invisibility also makes it a bit more fair (only kicks in when standing still) so you can set up ambushes but not actively stalk prey like a xeno hunter.

Dimishing returns on fleshmend is an interesting balance tweak, though not restoring lost blood is perhaps a bit heavy-handed. Maybe make it restore a very small amount (like 15u, aka one syringe worth) so it can help offset bleeding from wounds but require some serious spamming to replenish a significant amount of drained blood.

Do chems regenerate at normal speed while dead, or at a decreased rate? And instead of going to half capacity, why not just regen to whatever is necessary to regen?

**Comment by uraniummeltdown** [comment]
@FalseIncarnate 
15-20u of blood added seems fine, will change
Chems restore at normal speed while dead, I believe the reason they're capped at half is so that a freshly revived changeling can't immediately go on a rampage and armblade/shriek/sting everything, it'll be able to shriek or sting someone but if they want to use the heavy chem abilities they'll have to wait a bit

**Comment by Doukhan** [comment]
Changelings are OP already if you play them a certain way, but they rarely get very far anyway because their more useful and obvious abilities invariably get them lynched. These might be a bit much though especially with the armblade.
That said the changes to regenerative statis are much needed, I can't count the number of times I've lost a fight because I had to choose between sonic screeching NOW or being able to revive later.

**Comment by Birdtalon** [comment]
The chemical regeneration is very much needed.

**Comment by KingPhilipIII** [comment]
I still want to be a headless horseless horseman. Why can you even destroy changelings via decapitation. If they're based off The Thing, which every cell is it's own selfsustaining and selfpreserving component then something like destroying the brain doesn't really make sense. Plus the fact sec can just pop me onto any table and chop my head off upon capture(I've seen officers carry circular saws in their bags once I was going on a rampage and squads were hunting me down.) is a little annoying.

**Comment by Tayyyyyyy** [comment]
> When lings die, they will still generate chems up to half the maximum (so 37.5). I noticed that when lings got into fights, they'd use up all their chemicals fast, and when they finally died they couldn't go into Regen Stasis even if they had 14 chems. So now that they generate when dead you can go into Regen Stasis when you're dead.

I feel like the point of this is that you have to budget for regeneration. If you've used up all your chems in a fight, then too bad, you should've budgeted your chems better. There's sort of a trade off here: I can use all my chems to try to win this fight, but if I lose, then I won't be able to regen. Or I could play it safe and save some chems for regen. 

If regeneration is too expensive, then you should just lower the cost of regeneration. If you regen chems while dead, that basically makes the regeneration cost negligible and removes that part of strategizing when playing as changeling.

> I still want to be a headless horseless horseman. Why can you even destroy changelings via decapitation.

I think this would be pretty cool. A revived headless changeling would be blind so it's not exactly unbalanced. However, I think there are technical barriers to this. I believe the soul is connected to your brain, which is why decapitating ends the ling. Maybe if changelings had their brains in their chests, like IPCs, this would work.

**Comment by KingPhilipIII** [comment]
The thing about budgeting for regeneration is it doesn't actually have a point. If you're using your ling powers, if you lose the fight, it's over. By the time you regenerate ,if you even have that long, then you'll already be in the crematorium or on a surgical table or whatever. You better go all out because contrary to the design idea of the body must be destroyed or you rise, you only have to kill a changeling once and then slap a pair of cuffs on the corpse. You've pretty much won at that point because it's so easy to destroy them. Alternatively you can just do what you do with any other antagonist. Use a shotgun. If they even survive the initial blast their ribs are shattered and they can't fix them unless they go into stasis, at which point they are incredibly vulnerable and likely to be found and finished off, if they even stagger away.

**Comment by Tayyyyyyy** [comment]
> The thing about budgeting for regeneration is it doesn't actually have a point. If you're using your ling powers, if you lose the fight, it's over. By the time you regenerate ,if you even have that long, then you'll already be in the crematorium or on a surgical table or whatever.

Hmm okay, but then how does making chems regen when you're dead help? It seems like regen isn't useful in many cases because of meta knowledge of lings. If it's really that unuseful, just make regen free or super cheap.

**Comment by KingPhilipIII** [comment]
And then we come to our fundamental issue. Changelings are broken. Their whole gimmick is broken on a server that allows for antag knowledge, reducing them to a traitor with no gear but a few unique abilities and a built in radio. Or an alternate type of vampire. At their very core changelings are not unique. What would make them unique is pointless and ineffectual. All of their potential for roleplaying to make them unique is also out the window given the validhunty nature of our server, the play to win mentality and the general fact that instead of going "WHAT THE FUCK ARE YOU" it's "CHANGELING HELP ME SMASH ITS FACE IN WITH WHATEVER IS ON HAND."

Even their ability to assume identities is pointless except from the perspective of totally eliminate player from round or else they will come back and immediately snitch on you, blowing your cover. 

I like playing as antagonist but changeling is easily one of my least favorite.

**Comment by taukausanake** [comment]
> server that allows for antag knowledge

I think that's the main problem here. Pretty sure changelings were before all the other antags aside from traitors of course. They are the original, it's every other antag that's stepping on their feet

**Comment by KingPhilipIII** [comment]
Lings have not aged well then, unfortunantely. And they're stepping on a lot more than just their feet.

File: code/game/gamemodes/changeling/changeling_power.dm [hunk 1]
```diff
@@ -15,6 +15,7 @@
 	var/req_stat = CONSCIOUS // CONSCIOUS, UNCONSCIOUS or DEAD
 	var/genetic_damage = 0 // genetic damage caused by using the sting. Nothing to do with cloneloss.
 	var/max_genetic_damage = 100 // hard counter for spamming abilities. Not used/balanced much yet.
+	var/always_keep = 0 // important for abilities like regenerate that screw you if you lose them.
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/changeling_power.dm]: `FALSE`

**Pull request merged**